pkg/utils/meta_loader.go

```diff
@@ -56,9 +56,15 @@ func (l defaultMetaLoader) Load() (*addonsv1alpha1.AddonMetadataSpec, error) {
 		}
 		// deepcopy to be safe
 		meta.IndexImage = &imageSet.IndexImage
-		meta.AddOnParameters = imageSet.AddOnParameters.DeepCopy()
-		meta.AddOnRequirements = imageSet.AddOnRequirements.DeepCopy()
-		meta.SubOperators = imageSet.SubOperators.DeepCopy()
+		if meta.AddOnParameters != nil && imageSet.AddOnParameters != nil {
+			copy(*meta.AddOnParameters, *imageSet.AddOnParameters)
```

Comment:
Do we care if `meta.AddOnParameters != nil`? I think we **always** want to override whatever was there with the `imageSet.AddOnParameters` ? :thinking: 

:heart: nice refactor btw

Reply:
Maybe add a dummy unit test to `addons_test.go` ?

Reply:
> Do we care if meta.AddOnParameters != nil? I think we always want to override whatever was there with the imageSet.AddOnParameters ? 🤔

The problem is that if we don't perform a nil-check and straight away proceed with `copy` then, the `*meta.AddonParameters` and `*imageSet.AddOnParameters` trigger a panic due to nil dereference whenever one of those args is nil.

> // tested this code locally which returns empty dst

Wow! nice catch. Will fix this up, thanks :)

> Maybe add a dummy unit test to addons_test.go ?

Yep :)